### ISSUE TITLE
Incorrect changed result in cron module. 

### ISSUE THREAD

**Issue by chuggins** [ansible/ansible#4795]
The following task always results in a 'changed' message during plays:

``` yaml
- name: remove cron
  cron: cron_file=mycronfile state=absent
```

This appears to be due to this line in the cron module:

https://github.com/ansible/ansible/blob/devel/library/system/cron#L468

``` python
    if crontab.cron_file and not do_install:
        crontab.remove_job_file()
        changed = True # this line
        module.exit_json(changed=changed,cron_file=cron_file,state=state)
```

Changed needs to be set based on whether the `remove_job_file()` method actually unlinked a file or not.

**Comment by mpdehaan** [comment]
Since you've identified the above line, how about submitting a pull request?

Thanks!

**Comment by chuggins** [comment]
I could try to look in to it a bit later this week, though I haven't set up an ansible dev env yet.  Do you prefer patches against one of the current release branches, or devel?

**Comment by weaselkeeper** [comment]
I have a fix for this now if it's useful. As noted, not a tough one. I'll submit a PR if wanted.

**Comment by chuggins** [comment]
@weaselkeeper, thanks for that. I think the patch is missing a needed fix though, see commit for my comment.

**Comment by chuggins** [comment]
Looks great, thanks for putting that together.

**Comment by jctanner** [comment]
Closed per 7dac397d1afb6c024f6bbea8242268f282621b18
